oauthenticator/openshift.py

```diff
@@ -127,13 +142,42 @@ async def authenticate(self, handler, data=None):
         )
 
         resp = await http_client.fetch(req)
-        resp_json = json.loads(resp.body.decode('utf8', 'replace'))
+        ocp_user = json.loads(resp.body.decode('utf8', 'replace'))
 
-        return {
-            'name': resp_json['metadata']['name'],
-            'auth_state': {'access_token': access_token, 'openshift_user': resp_json},
+        username = ocp_user['metadata']['name']
+
+        user_info = {
+            'name': username,
+            'auth_state': {'access_token': access_token, 'openshift_user': ocp_user},
         }
 
+        if self.allowed_groups or self.admin_groups:
+            user_info = await self._add_openshift_group_info(user_info)
+
+        return user_info
+
+    async def _add_openshift_group_info(self, user_info: dict):
+        """
+        Use the group info stored on the OpenShift User object to determine if a user
+        is authenticated based on groups, an admin, or both.
+        """
+        user_groups = set(user_info['auth_state']['openshift_user']['groups'])
+
+        if self.admin_groups:
+            is_admin = self.user_in_groups(user_groups, self.admin_groups)
+
+        user_in_allowed_group = self.user_in_groups(user_groups, self.allowed_groups)
+
+        if self.admin_groups and (is_admin or user_in_allowed_group):
+            user_info['admin'] = is_admin
+            return user_info
+        elif user_in_allowed_group:
+            return user_info
+        else:
+            msg = "username:{username} User not in any of the allowed/admin groups"
```

Comment:
Supposed to be an fstring here right?

```suggestion
            msg = f"username:{username} User not in any of the allowed/admin groups"
```

Reply:
@consideRatio yes, good catch! fixed.